lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -820,6 +820,16 @@ def upload_grades_csv(_xmodule_instance_args, _entry_id, course_id, _task_input,
             course_grade.letter_grade,
             student.id in whitelisted_user_ids
         )
+        if certificate_info[0] == 'Y' and course_grade.letter_grade is None and student.id not in whitelisted_user_ids:
```

Comment:
not that this is going to be crazily computationally expensive hopefully, but since we're calculating `student.id in whitelisted_user_ids` twice (once in L821), maybe we should have a variable `student_is_whitelisted = student.id in whitelisted_user_ids`.

This is a bit of nit, so up to you if you want to implement.